lib/Re2.fx

```diff
@@ -0,0 +1,1234 @@
+/*
+    This file is a part of ficus language project.
+    See ficus/LICENSE for the licensing terms
+*/
+
+/*
+    Re2 library wrapper.
+
+    Usage: 
+    1.) Re2 library must be installed in system, see installation instructions there:
+    https://github.com/google/re2/wiki/Install
+
+    Sometimes(E.g.,, on Debian 10) it's also needed to run
+        sudo ldconfig 
+    after installation for creating appropriate soft links for shared libraries.
+
+    2.) Before compilation set environment variable FICUS_LINK_LIBRARIES to "-lre2" for providing compiler info about dependencies.
+        e.g, in bash:
+        export FICUS_LINK_LIBRARIES="-lre2"
+    3.) Use -c++ flag on compilation.
+
+
+    See https://github.com/google/re2/wiki/Syntax re2.h of Re2 source file for more detailed information about syntax and options. 
+    Wrapper function signatures have ficus-dependent specialization. They are described there.
+*/
+
+exception BadRegexp : string
+ccode 
+{
+    #include <re2/re2.h>
+    #include <vector>
+
+    int _fx_M3Re2FM14make_BadRegexpE1S(fx_str_t* arg0, fx_exn_t* fx_result); //[TODO] : Exception, declared before ccode block must be declared in program, translated to C program before this ccode for avoiding such insets. 
+
+    static int throwRe2Badregexp(const std::string& err)
+    {
+        fx_exn_t v_0 = {};
+        int fx_status = 0;
+        fx_str_t werr;
+
+        fx_status = fx_cstr2str(err.c_str(), err.size(), &werr);
+        if(fx_status >= 0) 
+        {
+            FX_CALL(_fx_M3Re2FM14make_BadRegexpE1S(&werr, &v_0), _fx_cleanup); //[TODO] : We need C-side infrastacture for user-registered exceptions. Name mangling for exception functions is a root of problem.
+            FX_THROW(&v_0, true, _fx_cleanup);
+        }
+        else
+        {
+            FX_UPDATE_BT();
+        }
+    _fx_cleanup: ;
+        fx_free_exn(&v_0);
```

Comment:
this call is not necessary, since the exception is *moved* into the 'currently thrown exception' global var, not copied. `FX_THROW()` always succeeds in throwing an exception, it does not allocate any memory etc.

Reply:
Done.